repos/system_upgrade/el7toel8/actors/addupgradebootentry/tests/unit_test.py

```diff
@@ -42,3 +43,9 @@ def consume_no_message_mocked(*models):
 
     with pytest.raises(StopActorExecutionError):
         library.get_boot_file_paths()
+
+
+def test_fix_grub_config_error():
+    library.fix_grub_config_error('files/grub_test.wrong')
```

Comment:
Crrently the test changes the test file. Ideally, unit tests should not be changing any file when executed. One solution is to monkeypatch file `write()` function.
It would be safer to use `monkeypatch.context()` [1] to make sure you don't break _pytest_ internals by monkeypatching `write()`.
[1] https://docs.pytest.org/en/latest/reference.html#_pytest.monkeypatch.MonkeyPatch.context

Reply:
Or, probably the easiest would be to create a file write wrapper and monkeypatch that:
```python
def write_to_file(filepath, content):
    with open(filepatch, 'w') as f:
        f.write(content)
```